pageserver/src/walingest.rs

```diff
@@ -304,8 +305,9 @@ impl<'a, R: Repository> WalIngest<'a, R> {
                 image.resize(image.len() + blk.hole_length as usize, 0u8);
                 image.unsplit(tail);
             }
-            image[0..4].copy_from_slice(&((lsn.0 >> 32) as u32).to_le_bytes());
-            image[4..8].copy_from_slice(&(lsn.0 as u32).to_le_bytes());
+            if !page_is_new(&image) {
```

Comment:
Please add a comment that this needs to match the logic in `XLogReadBufferForRedoExtended`

Reply:
Done